src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part1/index.md

```diff
@@ -0,0 +1,429 @@
+6. Create a RADOS snap pool with replication factor 3.
+7. Temporarily disable recovery and backfill ops.
+8. Bring up the downed OSD from step 5
+9. Prefill the snap pool with 70K 4MiB sized objects.
+10. Create a RADOS pool snapshot of the snap pool.
+11. Overwrite 100% of the objects in the snap pool. This creates 70K clone objects.
+12. Initiate I/O on the client pool for 300 secs using fio
+13. After client I/Os have stabilized for 30 secs, remove the snapshot created in step 10. This triggers the removal of the 70K clone objects as part of the snaptrim operation.
+14. At the same time, enable recovery and backfill ops (disabled in step 7). This triggers the backfill of misplaced objects.
+15. Statistics related to the client latency and throughput are captured. The test also captures the time taken to trim the 70K objects and time taken to recover the total misplaced objects.
+
+To summarize, the steps above create 3 pools during the test. Snaptrim, recovery and client ops are simultaneously triggered on separate pools. Statistics captured during the tests are discussed below.
+
+### Test Results
+
+#### Client Throughput Comparison
+
+Fig. 9 shows the average client throughput comparison across the schedulers and their respective configurations. The average baseline throughput with both the schedulers was determined to be around 23K IOPS (4KiB randrw) using fio on the client RBD pool.
+
+[![](images/Img9_Avg_Throughput_mClock_Profiles_vs_WPQ.png)](images/Img9_Avg_Throughput_mClock_Profiles_vs_WPQ.png)
+
+<center> Fig. 9: WPQ vs mClock Profiles - Average Client Throughput Comparison During Snaptrim+Recovery Ops</center>
+
+
+The client throughput shown is the average throughput reported by fio during the course of the test. It is important to note that both recovery/backfill and snaptrim durations form a subset of the client I/O run time.
+
+The average client throughput using the WPQ scheduler with default Ceph configuration was 12939.71 IOPS which is 43% lower than the baseline(WPQ) throughput. But with the mClock scheduler and with the default *high_client_ops* profile, the average client throughput is around 5.5% higher than the average throughput seen with WPQ scheduler at 13651.66 IOPS. This meets the minimum reservation criteria of 50% set for client ops in the mClock profile.
+
+The chart also shows the overall client throughput reported with the other mClock profiles. Due to lower reservations allocated to client ops in the other mClock profiles and with background recoveries and snaptrims active, significantly lower throughputs on the clients are expected. Therefore, with the *balanced* profile, an average throughput of 10541.26 IOPS was obtained. This is 45% of the baseline(mClock) throughput. Finally, with the *high_recovery_ops* profile, it was 9168.12 IOPS which is around 39% of the baseline(mClock) throughput. This shows that the minimum reservation criteria is being met for the mClock profiles.
```

Comment:
Oh, is the format of `baseline(mClock)` and `baseline(WPQ)` intentional? If so, disregard my suggestion to put in an extra space.

Reply:
Yes, it refers to the format used on the bar charts. I have therefore retained this format in the blog.